### PR TITLE
Abstract page - show more|less authors, affiliations, use layout file

### PR DESCRIPTION
**Pull request by shinyichen**
* option to show more|less authors
* option to show|hide  affiliations
* use layout files for the abs pages
Screen Shot 2021-09-29 at 3 15 13 PM

Screen Shot 2021-09-29 at 3 16 15 PM

### CODE REVIEW

File: src/pages/abs/[id]/abstract.tsx [hunk 1]
```diff
@@ -4,33 +4,102 @@ import { abstractPageNavDefaultQueryFields } from '@components/AbstractSideNav/m
...
+  const [showNumAuthors, setShowNumAuthors] = useState<number>(MAX_AUTHORS);
+
+  const [aff, setAff] = useState({ show: false, data: [] as string[] });
+
+  // onComponentDidMount
+  useEffect(() => {
+    if (showNumAuthors > doc.author.length) {
+      setShowNumAuthors(doc.author.length);
+    }
+  }, []);
```
Comment (thostetler) [inline src/pages/abs/[id]/abstract.tsx:40]: Doc should probably be a dep here, even though we'll be rerendering anyway

File: src/pages/abs/[id]/abstract.tsx [hunk 2]
```diff
@@ -4,33 +4,102 @@ import { abstractPageNavDefaultQueryFields } from '@components/AbstractSideNav/m
...
+  const handleShowAllAuthors = () => {
+    setShowNumAuthors(doc.author.length);
+  };
+
+  const handleShowLessAuthors = () => {
+    setShowNumAuthors(Math.min(doc.author.length, MAX_AUTHORS));
+  };
+
+  const handleShowAff = () => {
+    const adsapi = new AdsApi({ token: userData.access_token });
```
Comment (thostetler) [inline src/pages/abs/[id]/abstract.tsx:51]: You should switch to use the hook probably
```suggestion
    const { api } = useAPI();
```

File: src/pages/abs/[id]/abstract.tsx [hunk 3]
```diff
@@ -4,33 +4,102 @@ import { abstractPageNavDefaultQueryFields } from '@components/AbstractSideNav/m
...
+    if (aff.data.length === 0) {
+      params.fl = ['aff'];
+      void adsapi.search.query(params).then((res) => {
+        if (res.isOk()) {
```
Comment (thostetler) [inline src/pages/abs/[id]/abstract.tsx:55]: Make sure you take advantage of the `Result` it makes you handle any errors, you can use `.match()` to handle each case

**Review by thostetler — CHANGES_REQUESTED** [review]
Looks pretty good to me, a couple changes and then few free to merge

File: src/pages/abs/[id]/abstract.tsx (hunk 1 above)
Comment (shinyichen) [reply src/pages/abs/[id]/abstract.tsx]: Added doc as dependency

File: src/pages/abs/[id]/abstract.tsx (hunk 2 above)
Comment (shinyichen) [reply src/pages/abs/[id]/abstract.tsx]: Updated using suggested

File: src/pages/abs/[id]/abstract.tsx (hunk 3 above)
Comment (shinyichen) [reply src/pages/abs/[id]/abstract.tsx]: Using `.match()`

**Pull request merged**